demo/api/src/products/product-variant.service.ts

```diff
@@ -0,0 +1,19 @@
+import { type CrudGeneratorHooksService, type CurrentUser } from "@comet/cms-api";
+
+import { type ProductVariant } from "./entities/product-variant.entity";
+import { type ProductVariantInput, type ProductVariantUpdateInput } from "./generated/dto/product-variant.input";
+
+export class ProductVariantService implements CrudGeneratorHooksService {
+    async validateCreateInput(input: ProductVariantInput, options: { currentUser: CurrentUser; args: { product: string } }): Promise<void> {
```

Comment:
Is there any way we can improve the typing here?

Especially `args: { product: string }` seems cumbersome. But I suppose we need an args class to type this better